Source/JSONAPISerialization.swift

```diff
@@ -0,0 +1,164 @@
+    
+    private let data: AnyObject
+    
+    /**
+     Initialize a serializer with a single `JSONAPIEntity`
+     
+     - parameter object: A `JSONAPIEntities`
+     
+     - returns: A serializable object that serializes a `JSONAPIEntity` conforming to JSON API
+     */
+    public init(_ object: T) {
+        data = object.serialize()! // can't fail, JSONAPIEntity must always be serializable
+    }
+    
+    /**
+     Initialize a serializer with an array of `JSONAPIEntity`
+     
+     - parameter objects: An array of `JSONAPIEntity`
+     
+     - returns: A serializable object that serializes an array of `JSONAPIEntity` conforming to JSON API
+     */
+    public init(_ objects: [T]) {
+        data = objects.serialize()! // can't fail, JSONAPIEntity must always be serializable
+    }
+    
+    // MARK: CustomSerializable
+
+    public func customSerialize() -> AnyObject? {
+        return ["data": data]
+    }
+}
+
+// MARK: - Extensions
+
+extension Array: JSONAPISerializable {
+    
+    // MARK: JSONAPISerializable
+    
+    public func data(includeRelationships includeRelationships: Bool, includeAttributes: Bool) -> AnyObject? {
+        return Element.self is JSONAPISerializable.Type ? flatMap { ($0 as? JSONAPISerializable)?.data(includeRelationships: includeRelationships, includeAttributes: includeAttributes) } : nil
+    }
+}
+
+extension PropertyPolicy: JSONAPISerializable {
+
+    // MARK: JSONAPISerializable
+    
+    public func data(includeRelationships includeRelationships: Bool, includeAttributes: Bool) -> AnyObject? {
+        guard Value.self is JSONAPISerializable.Type else {
+            return nil
+        }
+        
+        switch self {
+        case let .Some(value):
+            if let value = value as? JSONAPISerializable {
+                return value.data(includeRelationships: includeRelationships, includeAttributes: includeAttributes)
+            }
+            
+        case .Null:
+            return [String: AnyObject]() // included as relationship but empty
+        case .None:
+            return nil
+        }
+        
+        return nil
+    }
+}
+
+public extension JSONAPIEntity {
+    
+    // MARK: JSONAPIEntity
+
+    var type: String {
+        return String(self.dynamicType).lowercaseString
+    }
+    
+    // MARK: CustomSerializable
+
+    public func customSerialize() -> AnyObject? {
+        return data(includeRelationships: true, includeAttributes: true)!
+    }
+    
+    // MARK: JSONAPISerializable
+
+    public func data(includeRelationships includeRelationships: Bool, includeAttributes: Bool) -> AnyObject? {
+        var data = [String: AnyObject]()
+        
+        data["id"] = id
+        data["type"] = type
+        
+        let mirror = Mirror(reflecting: self)
+        
+        var attributes = [String: AnyObject]()
+        var relationships = [String: AnyObject]()
+        
+        for child in mirror.children {
```

Comment:
We can avoid all the mirroring when `includeRelationships` or `includeAttributes ` are `false`